clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -1535,6 +1591,11 @@ int callbacks() {
         return invoker.callbackQueue.size();
     }
 
+    // Visible for testing
+    SubscriptionState subscriptions() {
+        return subscriptions;
+    }
```

Comment:
Is there a way to avoid this?

Reply:
- I can add callback to the constructor that registers the SubscriptionState with the calling context (the test)
- I can add another factory to inject our own instance of SubscriptionState